src/julia_internal.h

```diff
@@ -590,7 +590,108 @@ jl_tupletype_t *arg_type_tuple(jl_value_t *arg1, jl_value_t **args, size_t nargs
 
 int jl_has_meta(jl_array_t *body, jl_sym_t *sym);
 
-// backtraces
+//--------------------------------------------------
+// Backtraces
+
+// Backtrace buffers:
+//
+// A backtrace buffer conceptually contains a stack of instruction pointers
+// ordered from the inner-most frame to the outermost. We store them in a
+// special raw format for two reasons:
+//
+//   * Efficiency: Every `throw()` must populate the trace so it must be as
+//     efficient as possible.
+//   * Signal safety: For signal-based exceptions such as StackOverflowError
+//     the trace buffer needs to be filled from a signal handler where most
+//     operations are not allowed (including malloc) so we choose a flat
+//     preallocated buffer.
+//
+// The raw buffer layout contains "frame entries" composed of one or several
+// `jl_bt_element_t` values. From the point of view of the GC, an entry is
+// either:
+//
+// 1. A single instruction pointer to native code, not GC-managed.
+// 2. An "extended entry": a mixture of raw data and pointers to julia objects
+//    which must be treated as GC roots.
+//
+// An extended entry `e` is made up of several jl_bt_element_t values:
+//
+//   e[0]  JL_BT_NON_PTR_ENTRY  - Special marker to distinguish extended entries
+//   e[1]  tags                 - A bit packed uintptr_t containing a tag and
```

Comment:
Maybe call it `desc` / descriptor? I found it confusing how there's `tags` & `tag`.